src/test/java/org/alfresco/service/namespace/QNameTest.java

```diff
@@ -124,6 +124,211 @@ public void testInvalidQName() throws Exception
         
     }
 
+    public void testCreateQnameWithNoPrefixWithIllegalCharacters()
+    {
+        char[] illegalCharacters = {'/', '\\', '\n', '\r', '"'};
+        for (char illegalCharacter : illegalCharacters)
+        {
+            String localName = "testLocalNameWith" + illegalCharacter;
+            try
+            {
+                QName.createQName(NamespaceService.ALFRESCO_PREFIX, localName);
+            }
+            catch (InvalidQNameException e)
+            {
+                assertEquals(e.getMessage(), "name contains characters that are not permitted: "+localName.charAt(localName.indexOf(illegalCharacter)));
+            }
+        }
+    }
+
+    public void testCreateQnameWithNoPrefixWithNullAndEmptyCharacters()
```

Comment:
There is a lot of duplication in the tests. This should be avoided.
A cleaner way to test would be something like this:
```
    @Test(expected = InvalidQNameException.class)
    public void testCreateQnameWithNoPrefixWithNull()
    {
        String nullLocalName = null;
        QName.createQName(NamespaceService.ALFRESCO_PREFIX, nullLocalName);
    }
```

Reply:
I noticed the existing tests are written using JUnit3 and the use of annotation `@Test(expected = InvalidQNameException.class)` is not working as expected :(

I've read something related to this:
> JUnit4 style tests should not extend junit.framework.TestCase directly or indirectly. If you do, you often get confusing behavior (fields annotated by @Rule ignored, test failures because the base class methods are not being called by JUnit, etc). The tests may be run as JUnit4-style or JUnit3-style depending bon how they are run.
> 
> Your options are either to stick with JUnit3 or convince the maintainers of your base class to create classes that support JUnit4 (Rules, Runners, etc).

Reply:
However I will try to reduce the noise by using a simpel `try/catch` method to only capture the exception (like other tests).

Reply:
After removing the `extends TestCase` and removing the public constructor method that was causing problems in using `@Test` annotation, the Tests are now refactored to just catch the expected exception.